assets/js/modules/analytics-4/datastore/custom-dimensions-gathering-data.js

```diff
@@ -0,0 +1,395 @@
+
+			state.customDimensionsDataAvailableOnLoad[ customDimension ] =
+				dataAvailableOnLoad;
+
+			break;
+		}
+
+		default: {
+			break;
+		}
+	}
+} );
+
+const baseResolvers = {
+	*isCustomDimensionGatheringData( customDimension ) {
+		const registry = yield Data.commonActions.getRegistry();
+
+		// If the gatheringData flag is already set, return early.
+		if (
+			registry
+				.select( MODULES_ANALYTICS_4 )
+				.isCustomDimensionGatheringData( customDimension ) !== undefined
+		) {
+			return;
+		}
+
+		const dataAvailableOnLoad = registry
+			.select( MODULES_ANALYTICS_4 )
+			.isCustomDimensionDataAvailableOnLoad( customDimension );
+
+		// If dataAvailableOnLoad is true, set gatheringData to false and do nothing else.
+		if ( dataAvailableOnLoad ) {
+			yield baseActions.receiveIsCustomDimensionGatheringData(
+				customDimension,
+				false
+			);
+			return;
+		}
+
+		yield {
+			payload: {
+				customDimension,
+			},
+			type: WAIT_FOR_CUSTOM_DIMENSION_DATA_AVAILABILITY_STATE,
+		};
+
+		const dataAvailability = registry
+			.select( MODULES_ANALYTICS_4 )
+			.selectCustomDimensionDataAvailability( customDimension );
+
+		yield baseActions.receiveIsCustomDimensionGatheringData(
+			customDimension,
+			! dataAvailability
+		);
+
+		if ( dataAvailability ) {
+			yield fetchSaveCustomDimensionDataAvailableStateStore.actions.fetchSaveCustomDimensionDataAvailableState(
+				customDimension
+			);
+		}
+	},
+};
+
+const baseSelectors = {
+	/**
+	 * Determines whether data is available for the custom dimension.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state           Data store's state.
+	 * @param {string} customDimension Custom dimension slug.
+	 * @return {boolean|undefined|null} Returns TRUE if data is available, otherwise FALSE.
+	 *                                  If the request is still being resolved, returns undefined.
+	 *                                  If the data availability can not be determined, returns null.
+	 */
+	selectCustomDimensionDataAvailability: createRegistrySelector(
+		( select ) => ( state, customDimension ) => {
+			if (
+				! select( MODULES_ANALYTICS_4 ).hasCustomDimensions(
+					customDimension
+				)
+			) {
+				return false;
+			}
+
+			const isAuthenticated = select( CORE_USER ).isAuthenticated();
+
+			if ( isAuthenticated === undefined ) {
+				return undefined;
+			}
+			if ( ! isAuthenticated ) {
+				return false;
+			}
+
+			const propertyID = select( MODULES_ANALYTICS_4 ).getPropertyID();
+
+			if ( propertyID === undefined ) {
+				return undefined;
+			}
+
+			const property =
+				select( MODULES_ANALYTICS_4 ).getProperty( propertyID );
+
+			if ( property === undefined ) {
+				return undefined;
+			}
+
+			const startDate = getDateString( new Date( property.createTime ) );
+
+			const endDate = select( CORE_USER ).getReferenceDate();
+
+			const reportArgs = {
+				startDate,
+				endDate,
+				dimensions: [ `customEvent:${ customDimension }` ],
+			};
```

Comment:
It won't hurt if we add a metric for this report (something like `eventCount`), but having a metric here will let us avoid making additional changes that you added to allow reports without metrics. Let's better change this report to have a basic metric.

Also, we can limit the number of rows returned. We can use `limit: 2` here to make sure we don't pull excessive data.

Reply:
Thanks, `limit: 2` is definitely a good shout. My thinking with regard to omitting the metric is that I would expect it to be a slightly lesser load on the GA4 service for this report, which seems like a general good practice to keep up. I'm interested in your thoughts with this in mind.

Reply:
Limiting the number of rows to return helps more than omitting the metric. With just two rows returned, having a metric doesn't add a huge load. So, I think we should revert changes that allow having no metrics for reports and use a basic metric for this report.

Reply:
Thanks @eugene-manuilov - I've updated the report to use `eventCount` as suggested and rolled back the unneeded changes.